lib/nostrum/shard/event.ex

```diff
@@ -3,13 +3,10 @@ defmodule Nostrum.Shard.Event do
 
   alias Nostrum.Shard.Payload
   alias Nostrum.Shard.Stage.Producer
-  alias Nostrum.Util
 
   require Logger
 
   def handle(:dispatch, payload, state) do
-    payload = Util.safe_atom_map(payload)
```

Comment:
IIRC this will definitely at the very least cause breaking changes.

Anyone who attempts to access a key by via the atom syntax (`guild.id` for example) will be at risk of having issues appear. When I was implementing this functionality I was looking at the guild response I think and in that struct alone there were differing keys.

I think if we wanted to do this piecemeal we'd need to go into the event handlers (on our side) and change them at a high level there to convert to atoms, but add a header for this problematic payload for now that does not convert them. Thoughts?

Reply:
I took another look and I see what you're talking about now with this probably not being an issue because of how we build the structs. If we do offer any events not as a struct though it would still be an issue, and I believe we do this in a lot of places.

Reply:
Yes, there are quite a few, via `lib/nostrum/shard/dispatch.ex`. Should I convert those to structs?

Reply:
I think ideally they should be yes. But I also know it's a lot of work. :(

Reply:
Yes, it is. "Good things take time", or something like that.

We could also `safe_atom_map` those instead. But I vouch for the clean solution.

Reply:
This should be taken care of by https://github.com/Kraigie/nostrum/commit/1674954530b7ef96428aa0484d1ec99d9ade714c.